### PR TITLE
93 pair remove redundant slippage check during providing liquidity

### PR DESCRIPTION
**Pull request by gangov**

### CODE REVIEW

File: contracts/pair/src/contract.rs [hunk 1]
```diff
@@ -211,6 +219,7 @@ impl LiquidityPoolTrait for LiquidityPool {
                     min_b,
                     pool_balance_a,
                     pool_balance_b,
+                    Decimal::bps(custom_slippage_bps.unwrap_or(1)),
```
Comment (ueco-jb) [inline contracts/pair/src/contract.rs:222]: https://www.investopedia.com/terms/b/basispoint.asp
1 BPS is not a 1%.

File: contracts/pair/src/storage.rs [hunk 2]
```diff
@@ -402,7 +443,16 @@ mod tests {
...
-        let result = utils::get_deposit_amounts(&env, 5000, Some(2000), 2000, None, 10000, 5000);
+        let result = utils::get_deposit_amounts(
+            &env,
+            5000,
+            Some(2000),
+            2000,
+            None,
+            10000,
+            5000,
+            Decimal::bps(1),
```
Comment (ueco-jb) [inline contracts/pair/src/storage.rs:454]: As above.

File: contracts/pair/src/storage.rs [hunk 3]
```diff
@@ -413,7 +463,16 @@ mod tests {
...
-        let result = utils::get_deposit_amounts(&env, 5000, None, 5000, Some(1000), 5000, 10000);
+        let result = utils::get_deposit_amounts(
+            &env,
+            5000,
+            None,
+            5000,
+            Some(1000),
+            5000,
+            10000,
+            Decimal::bps(1),
```
Comment (ueco-jb) [inline contracts/pair/src/storage.rs:474]: And here.

File: contracts/pair/src/storage.rs [hunk 4]
```diff
@@ -332,57 +332,97 @@ mod tests {
     #[test]
     fn test_get_deposit_amounts_pool_balances_zero() {
         let env = Env::default();
-        let result = utils::get_deposit_amounts(&env, 100, Some(50), 200, Some(50), 0, 0);
+        let result =
+            utils::get_deposit_amounts(&env, 100, Some(50), 200, Some(50), 0, 0, Decimal::one());
```
Comment (ueco-jb) [inline contracts/pair/src/storage.rs:336]: `Decimal::one` is 1.0 meaning 100%. In original test it was 1%.

**Review by ueco-jb — COMMENTED** [review]
Changeset is good, but you assigned original values in test as way too high or too low.

**Review by ueco-jb — APPROVED** [review]
You can merge after modifying those values.

File: contracts/pair/src/contract.rs (hunk 1 above)
Comment (gangov) [reply contracts/pair/src/contract.rs:222]: that explains it greatly, thanks!

**Comment by gangov** [comment]
I see what you wrote, but I'd feel better if you give it a 2nd quick look

**Review by ueco-jb — APPROVED** [review]
If you are unsure about using BPSs, you can as well use `Decimal::percent` in tests, might be more intuitive.
But changes are correct, you can merge it. :+1:

**Pull request merged**